kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Coordinator.java

```diff
@@ -215,16 +222,31 @@ private void commitToTable(
       LOG.info("Nothing to commit to table {}, skipping", tableIdentifier);
     } else {
       if (deleteFiles.isEmpty()) {
-        AppendFiles appendOp = table.newAppend();
-        branch.ifPresent(appendOp::toBranch);
-        appendOp.set(snapshotOffsetsProp, offsetsJson);
-        appendOp.set(COMMIT_ID_SNAPSHOT_PROP, commitState.currentCommitId().toString());
-        if (vtts != null) {
-          appendOp.set(VTTS_SNAPSHOT_PROP, Long.toString(vtts));
-        }
-        dataFiles.forEach(appendOp::appendFile);
-        appendOp.commit();
+        Map<Integer, List<DataFile>> dataFilesBySpec = new HashMap<>();
+        dataFiles.forEach(dataFile -> {
+          List<DataFile> list = dataFilesBySpec.computeIfAbsent(dataFile.specId(), ignored -> Lists.newArrayList());
+          list.add(dataFile);
+        });
+        Integer maxSpecId = dataFilesBySpec.keySet().stream().max(Integer::compare).get();
+
+        Transaction transaction = table.newTransaction();
+        dataFilesBySpec.entrySet().stream().sorted(Comparator.comparingInt(Map.Entry::getKey)).forEach(entry -> {
+          AppendFiles appendOp = transaction.newAppend();
+          branch.ifPresent(appendOp::toBranch);
+          entry.getValue().forEach(appendOp::appendFile);
+          appendOp.set(COMMIT_ID_SNAPSHOT_PROP, commitState.currentCommitId().toString());
+          if (Objects.equals(entry.getKey(), maxSpecId)) {
```

Comment:
We only include the `controlTopicOffsets` and `vtts` properties in the last snapshot to be appended as part of this transaction. That's because we can only really be sure that these values are valid for the last snapshot to be appended.